go/teams/service_helper.go

```diff
@@ -687,38 +686,37 @@ func RemoveMember(ctx context.Context, g *libkb.GlobalContext, teamname, usernam
 		err = nil
 	}
 
+	me, err := loadMeForSignatures(ctx, g)
+	if err != nil {
+		return err
+	}
+
+	if me.GetNormalizedName().Eq(libkb.NewNormalizedUsername(username)) {
+		return Leave(ctx, g, teamname, false)
+	}
+
 	return RetryOnSigOldSeqnoError(ctx, g, func(ctx context.Context, _ int) error {
 		t, err := GetForTeamManagementByStringName(ctx, g, teamname, true)
 		if err != nil {
 			return err
 		}
 
 		if inviteRequired && !uv.Uid.Exists() {
-			// This branch only handles social invites. Keybase-type
-			// invites are handled by next removeMemberInvite call below.
+			// Remove a social (non-keybase) invite.
 			return removeMemberInvite(ctx, g, t, username, uv)
 		}
 
-		existingUV, err := t.UserVersionByUID(ctx, uv.Uid)
-		if err != nil {
-			// Try to remove as an keybase-invite
-			if ierr := removeKeybaseTypeInviteForUID(ctx, g, t, uv.Uid); ierr == nil {
-				return nil
-			}
-			return libkb.NotFoundError{Msg: fmt.Sprintf("user %q is not a member of team %q", username,
-				teamname)}
+		if _, _, found := t.FindActiveKeybaseInvite(uv.Uid); found {
+			// Remove keybase invites.
+			return removeKeybaseTypeInviteForUID(ctx, g, t, uv.Uid)
```

Comment:
The semantics here seem a bit weird - `FindActiveKeybaseInvite` only looks for active invites, but `removeKeybaseTypeInviteForUID` will remove either active or obsolete invite from the chain. This seems to make sense in this context, but was that something that you considered?

I wonder if calling `removeKeybaseTypeInviteForUID` alone and "falling through" on `libkb.NotFoundError` could also work - but this would change the logic slightly wrt obsoleted invites.

Reply:
in grand scheme of things it probably doesn't matter since we try hard not to leave any obsolete invites anymore, but there might be a number of them from older, simpler, times.

I'm :+1:  in general but I want to know what you think. Thanks!

Reply:
I didn't consider that. At this point it seems like it's hard to prove invariants so paths have to be able to function on large input spaces.

Reply:
no worries, I spent some time trying to find a test case where it would break but couldn't get a simple one.